index.js

```diff
@@ -1,26 +1,12 @@
-const merge = require('webpack-merge')
-const { applyConfig } = require('./lib/config-helpers')
 const { join } = require('path')
-
-const defaultSharedConfig = require('./webpack.shared.config')
-const defaultClientConfig = require('./webpack.client.config')
-const defaultServerConfig = require('./webpack.server.config')
-const defaultDevConfig = require('./webpack.dev.config')
+const configMerge = require('./lib/configMerge')
```

Comment:
I kinda thought this would go into `lib/build` and `lib/serve`, rather than in here.

We can also then leave the destructuring of all args to each of those functions:

Instead of:

```
// ./index.js

module.exports = ({
  action,
  someRelevantArg,
  someIrrelevantArg
}) => (
  if (action === 'foo') {
    foo({ someRelevantArg })
  }
)
```

You could do:

```
// ./index.js

module.exports = ({ action, ...opts }) => actions[action](opts)

// ./lib/foo.js

module.exports = ({ someRelevantArg }) => { /* Profit */ }
```